src/core/services/OnChainMetaDataService.ts

```diff
@@ -0,0 +1,20 @@
+import { _getContractABI } from '@src/utils';
+import { OnChainMetaDataService, Web3Provider } from '@types';
+import {} from '@ethersproject/address';
+
+export class OnChainMetaDataServiceImpl implements OnChainMetaDataService {
+  private web3Provider: Web3Provider;
+  constructor({ web3Provider }: { web3Provider: Web3Provider }) {
+    this.web3Provider = web3Provider;
+  }
+
+  public async getContractABI(address: String, network: number): Promise<any | undefined> {
+    return await _getContractABI(address, network);
+  }
+
+  public async getAddressEnsName(address: string): Promise<any | undefined> {
+    const provider = this.web3Provider.getInstanceOf('ethereum');
+    const addressEnsName = await provider.lookupAddress(address);
```

Comment:
possible to batch these calls if we want to?

Reply:
``

 public async getAddressEnsName(address: string): Promise<string | null> {
    const addressEnsName = await this.web3Provider.getInstanceOf('ethereum').lookupAddress(address);
    return addressEnsName;
  }
  
  ``